utils/merge_syft_sbom.py

```diff
@@ -118,6 +118,29 @@ def component_is_duplicated(component: dict[str, Any]) -> bool:
     return component_is_duplicated
 
 
+def _merge_tools_metadata(syft_sbom: dict[Any, Any], cachi2_sbom: dict[Any, Any]) -> None:
+    """Merge the content of tools in the metadata section of the SBOM.
+
+    With CycloneDX 1.5, a new format for specifying tools was introduced, and the format from 1.4
+    was marked as deprecated.
+
+    This function aims to support both formats in the Syft SBOM. We're assuming the Cachi2 SBOM
+    was generated with the same version as this script, and it will be in the newer format.
```

Comment:
I don't think you should assume that. The SBOM is generated in the prefetch task, the merging happens in the build task. It's up to the user whether the task versions will be in sync or not.

Reply:
New approach explained [here](https://github.com/containerbuildsystem/cachi2/pull/448#issuecomment-1946179822).